docs/DeveloperGuide.md

```diff
@@ -1,11 +1,233 @@
+
+### UI component
+
+### Logic component
+
+### Model component
+
+### Storage component
+
+In this application, it uses text files to store all data, i.e, recipes and ingredients. 
+
+
+When booting up the application, `restoreSavedData` will be called to get both recipes and ingredients 
+from the previous session.
+
+<img src="images/RestoreIngredientStorageSequenceDiagram.png" width="347" />
+<img src="images/RestoreRecipeStorageSequenceDiagram.png" width="347" />
+
+When a user exits the application, `saveData` will be called and it will convert all data from recipes and
+ingredients into a string which will be put in their own respective text files.
+
+<img src="images/StoreIngredientStorageSequenceDiagram.png" width="525" />
+<img src="images/StoreRecipeStorageSequenceDiagram.png" width="706" />
+
+
+## Implementation
+### Help Feature
+The help feature is facilitated by the `HelpCommand` class. By calling `executeCommand` on the class, it will invoke the `Ui` class to print the user help commands.
+
+
+- **Step 1**
+
+  Input will be sent from the main `EssenMakanan` class to the `Parser` to identify the command type
+
+
+- **Step 2**
+
+  A new `HelpCommand` object will be created and will be sent back to main
+
+
+- **Step 3**
+
+  `commandObject#executeCommand()` will be called which in turn calls `Ui#showCommands()`
+
+
+- **Step 4**
+
+  Finally `Ui#showCommands()` will call `Ui#showRecipeCommands()`, `Ui#showIngredientCommands()`, `Ui#showOtherCommands()` to print all commands for recipe, ingredient and others respectively
+
+
+<img src="images/HelpFunctionSequenceDiagram.png" width="732" />
+
+### Exit feature
+The help feature is facilitated by the `ExitCommand` class. By calling `executeCommand` on the class, it will invoke the `Ui` class to print the exit command.
+
+- **Step 1**
+
+  Input will be sent from the main `EssenMakanan` class to the `Parser` to identify the command type
+
+
+- **Step 2**
+
+  A new `ExitCommand` object will be created and will be sent back to main
+
+
+- **Step 3**
+
+  `commandObject#executeCommand()` will be called which in turn calls `Ui#showCommands()`
+
+
+- **Step 4**
+
+  Finally `Ui#showCommands()` will print the exit message
+
+
+<img src="images/ExitSequenceDiagram.png" width="571" />
+
+
+### Add Recipe feature
+
+The add recipe feature is facilitated by the `AddRecipeCommand` class. By calling `executeCommand` on the class, the steps will
+be executed as follows:
+- **Step1**
+  
+  `AddRecipeCommand` will parse the recipe title using `RecipeParser`. Then, it will return the recipe title.
+
+
+- **Step2**
+
+  `AddRecipeCommand` will create a new `Recipe` with the obtained title.
+
+
+- **Step3**
+
+  `AddRecipeCommand` will add newly created `Recipe` into `RecipeList`. Then, the recipe will be added into an
+  `ArrayList` inside `RecipeList`.
+
+
+- **Step4**
+
+  `AddRecipeCommand` will call `Ui` class to print out the title of the recently added recipe.
```

Comment:
Maybe for the addRecipeCommand, you can consider explaining how the title parsing and RecipeList work in more detail. Provide a step-by-step breakdown of the process, including the role of RecipeParser and RecipeList.